src/librbd/cache/pwl/ssd/WriteLog.cc

```diff
@@ -666,9 +666,21 @@ bool WriteLog<I>::retire_entries(const unsigned long int frees_per_tx) {
             this->m_blocks_to_log_entries.remove_log_entry(write_entry);
           }
         }
+
+        ldout(cct, 20) << "span with " << retiring_subentries.size()
+                       << " entries: control_block_pos=" << control_block_pos
+                       << " data_length=" << data_length
+                       << dendl;
         retiring_entries.insert(
             retiring_entries.end(), retiring_subentries.begin(),
             retiring_subentries.end());
+
+        first_valid_entry = control_block_pos + data_length +
```

Comment:
For https://github.com/ceph/ceph/pull/42843/commits/2d337fb122d147e32d027d1e7211cd4156a5b72b#R632 this reason,  controlc_block_pos dont' equal retiring_subentries.back()->log_entry_index.

Reply:
Are you referring to when it encounters a log entry that belongs to the next span?  Maybe I'm missing something but in that case `control_block_pos` doesn't get modified.  We break out of the `for` loop and if `retiring_subentries` is not empty add them to `retiring_entries` -- both `control_block_pos` and `data_length` describe the current span.

Could you elaborate?